ethcore/src/client/test_client.rs

```diff
@@ -372,8 +490,8 @@ impl MiningBlockChainClient for TestBlockChainClient {
 	fn prepare_open_block(&self, author: Address, gas_range_target: (U256, U256), extra_data: Bytes) -> OpenBlock {
 		let engine = &*self.spec.engine;
 		let genesis_header = self.spec.genesis_header();
-		let mut db_result = get_temp_state_db();
-		let db = self.spec.ensure_db_good(db_result.take(), &Default::default()).unwrap();
+        let db_result = self.state_db.lock().boxed_clone();
```

Comment:
Indentation

Reply:
That's a quite different logic. Currently it re-uses the same state, previously it was creating fresh state each time, no?